include/nbl/video/SPhysicalDeviceLimits.h

```diff
@@ -0,0 +1,497 @@
+#ifndef _NBL_VIDEO_S_PHYSICAL_DEVICE_LIMITS_H_INCLUDED_
+#define _NBL_VIDEO_S_PHYSICAL_DEVICE_LIMITS_H_INCLUDED_
+
+#include <type_traits>
+#include "nbl/asset/utils/IGLSLCompiler.h" // asset::IGLSLCompiler::E_SPIRV_VERSION
+
+
+namespace nbl::video
+{
+
+struct SPhysicalDeviceLimits
+{
+    /* Vulkan 1.0 Core  */
+    uint32_t maxImageDimension1D;
+    uint32_t maxImageDimension2D;
+    uint32_t maxImageDimension3D;
+    uint32_t maxImageDimensionCube;
+    uint32_t maxImageArrayLayers;
+    uint32_t maxBufferViewSizeTexels;
+    uint32_t maxUBOSize;
+    uint32_t maxSSBOSize;
+    //uint32_t              maxPushConstantsSize;
```

Comment:
please report, but on OpenGL we've capped it to 128 bytes (check if we didn't hardcode anything else in the engine to 128 bytes - except IMeshBuffer's PC storage)

Reply:
Yea it's by checking `IGPUMeshBuffer::MAX_PUSH_CONSTANT_BYTESIZE`
I searched for this keyword and It doesn't seem like it's capped anything else other than this and MeshBuffer PC stuff like `SGLTFMaterialParameters` or `SMtl::params`

Reply:
I think `COpenGLCommandBuffer` and a few other classes disagree.

Basically I don't know about Vulkan, but on GL you'll always report `MAX_PUSH_CONSTANT_BYTESIZE` (128)  due to the hardcoding.